pkg/kubectl/cmd/create_configmap.go

```diff
@@ -77,18 +79,21 @@ func CreateConfigMap(f *cmdutil.Factory, cmdOut io.Writer, cmd *cobra.Command, a
 		return err
 	}
 	var generator kubectl.StructuredGenerator
+	var resource unversioned.GroupVersionResource
 	switch generatorName := cmdutil.GetFlagString(cmd, "generator"); generatorName {
 	case cmdutil.ConfigMapV1GeneratorName:
 		generator = &kubectl.ConfigMapGeneratorV1{
 			Name:           name,
 			FileSources:    cmdutil.GetFlagStringSlice(cmd, "from-file"),
 			LiteralSources: cmdutil.GetFlagStringSlice(cmd, "from-literal"),
 		}
+		resource = v1.SchemeGroupVersion.WithResource("configmap")
```

Comment:
> I think we want to set this to a partial GroupVersionResource (one that doesn't specify a version, like api.Resource(...).WithVersion("")). That matches what we're actually producing (an api.ConfigMap), and lets restmapper do partial matches to find the version it should create. @deads2k, does that sound right to you?

That really depends on your view of generators.  Given my view of them, yes.  Given how I've seen them pitched, no.

If generators are supposed to fulfill an intent by whatever means they deem appropriate based on negotiated level, then yes autodetection makes sense.   If generators are supposed to give stable output, they should always deal in external versions and should never be converted so its inappropriate to auto-detect the version.